src/NuGet.Core/NuGet.Protocol/HttpSource/HttpRetryHandler.cs

```diff
@@ -71,16 +78,21 @@ internal HttpRetryHandler(IEnvironmentVariableReader environmentVariableReader)
             }
 
             var tries = 0;
-            HttpResponseMessage response = null;
+            HttpResponseMessage? response = null;
             var success = false;
+            TimeSpan? retryAfter = null;
 
             while (tries < request.MaxTries && !success)
             {
                 // There are many places where another variable named "MaxTries" is set to 1,
                 // so the Delay() never actually occurs.
                 // When opted in to "enhanced retry", do the delay and have it increase exponentially where applicable
                 // (i.e. when "tries" is allowed to be > 1)
-                if (tries > 0 || (_enhancedHttpRetryHelper.IsEnabled && request.IsRetry))
+                if (retryAfter != null && _enhancedHttpRetryHelper.ObserveRetryAfter)
+                {
+                    await Task.Delay(retryAfter.Value, cancellationToken);
```

Comment:
If `retyAfter.Value` set to very large number due to large number like 1 hours for some reason, then should we wait for 1 hour.  For example, machine local datetime is off by 1 hour.

Reply:
Isn't this the same topic as https://github.com/NuGet/NuGet.Client/pull/4912#discussion_r1019675339?  Let's not have two threads for the same topic.